File: samples/dotnet/KernelHttpServer/Model/Ask.cs
// Copyright (c) Microsoft. All rights reserved.

using System.Collections.Generic;
using System.Linq;

namespace KernelHttpServer.Model;

public class Ask
{
    public string Value { get; set; } = string.Empty;

    /// <summary>
    /// When a non-empty collection is given, only skills matching these names should be loaded
    /// </summary>
    public IEnumerable<string>? Skills { get; set; } = null;

    public IEnumerable<AskInput> Inputs { get; set; } = Enumerable.Empty<AskInput>();
}

public class AskInput
{
    public string Key { get; set; } = string.Empty;

    public string Value { get; set; } = string.Empty;
}

File: samples/python/kernel_http_server/completion/ask.py
from dataclasses import dataclass
from typing import List


@dataclass
class AskInput:
    key: str = None
    value: str = None


@dataclass
class Ask:
    skills: List[str] = None
    inputs: List[AskInput] = None
    value: str = None


@dataclass
class AskResult:
    value: str = None
    state: List[AskInput] = None
